src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -348,10 +348,15 @@ public void notifyOfNewStep(@NonNull Step step, @NonNull StepContext context) {
                 return;
             }
             if (computer.getAction(OpenTelemetryAttributesAction.class) == null) {
-                LOGGER.log(Level.WARNING, "Unexpected missing " + OpenTelemetryAttributesAction.class + " on " + computer + " fallback");
+                LOGGER.log(Level.WARNING, "Unexpected missing " + OpenTelemetryAttributesAction.class + " on " + computer + ", adding fallback");
                 String hostName = computer.getHostName();
                 OpenTelemetryAttributesAction openTelemetryAttributesAction = new OpenTelemetryAttributesAction();
-                openTelemetryAttributesAction.getAttributes().put(ResourceAttributes.HOST_NAME, hostName);
+                if (hostName != null) {
+                    // getHostName() returns null if the master cannot find the host name, e.g. due to network settings.
+                    // @see hudson.model.Computer#getHostName()
+                    openTelemetryAttributesAction.getAttributes().put(ResourceAttributes.HOST_NAME, hostName);
+                }
+                openTelemetryAttributesAction.getAttributes().put(AttributeKey.stringKey(JenkinsOtelSemanticAttributes.JENKINS_COMPUTER_NAME.getKey()), computer.getName());
```

Comment:
this value is already set https://github.com/jenkinsci/opentelemetry-plugin/blob/master/src/main/java/io/jenkins/plugins/opentelemetry/computer/MonitoringComputerListener.java#L123

Reply:
Yes, the `MonitoringComputerListener` adds an `OpenTelemetryAttributesAction` on the `Computer` object. But that doesn't always happen! I know that because I traced an error to this `if (computer.getAction(OpenTelemetryAttributesAction.class) == null)` block. Previous authors probably knew this because this if block already existed.

I don't know when/why this can happen, but when it does, `JENKINS_COMPUTER_NAME` is not set on the step span. But the computer name is always available (unlike the hostname), so I add it here.

I could move the creation of `Computer` attributes to a separate function, and then call it in both places: always in `MonitoringComputerListener.preOnline`, and in `MonitoringPipelineListener.notifyOfNewStep` only when the `Computer` doesn't already have the Action attached.

That approach has the risk that, if calculating the attributes takes a long time or errors out, it will happen again and again; this could be mitigated if the code is robust and just adds an empty (or best-effort) Action in those cases. What do you think?